### ISSUE TITLE
Node Attr() calls lose request information

### ISSUE THREAD

**Issue by sjpotter** [bazil/fuse#252]
Node.Attr() is called from one place - server.go's nodeAttr()

nodeAttr() is called from 2 places, serverNode.attr() and Server.saveLookup()

both of those function are only called from Server.handleRequest()

hence, all calls to Attr() have to come from a request.  however, because of the API, this information is lost.  While I can satisfy the NodeGet/Setattrer interfaces, that doesn't handle any of the case from saveLookup().

basically, for the 6 cases from saveLookup()

        SymlinkRequest
        LinkRequest
        LookupRequest
        MkdirRequest
        CreateRequest
        MknodRequest

I'd like to be able to trace why I'm being called.  while I can call fstestutil.DebugByDefault() that is too verbose for my needs, I can easily stick in my own tracing functions that can be turned on / off as needed with a better level of granularity.

**Comment by tv42** [comment]
I think this will be done via making Attr more optional, and letting you do your attribute filling directly in those locations, with the tracing you already get. I think keeping Attr is better than adding a phony AttrRequest concept, or trying to cram all those requests into an interface argument.

In that sense, I think this can be called a duplicate of https://github.com/bazil/fuse/issues/109 -- your use case is different, but that answer should be the right thing for you too.
